src/os/ui/file/exportstatus.js

```diff
@@ -0,0 +1,182 @@
+
+
+/**
+ * Add the directive to the module.
+ */
+Module.directive(directiveTag, [directive]);
+
+
+/**
+ * Controller for the exportstatus directive.
+ * @unrestricted
+ */
+export class Controller {
+  /**
+   * Constructor.
+   * @param {!angular.Scope} $scope The Angular scope.
+   * @param {!angular.JQLite} $element The root DOM element.
+   * @ngInject
+   */
+  constructor($scope, $element) {
+    /**
+     * The Angular Scope.
+     * @type {?angular.Scope}
+     * @protected
+     */
+    this.scope = $scope;
+
+    /**
+     * The Angular element.
+     * @type {?angular.JQLite}
+     * @protected
+     */
+    this.element = $element;
+
+    /**
+     * The exporter.
+     * @type {IExportMethod}
+     * @protected
+     */
+    this.exporter = /** @type {IExportMethod} */ ($scope['exporter']);
+
+    const items = this.exporter.getItems();
+    /**
+     * The total feature count.
+     * @type {number}
+     * @protected
+     */
+    this['total'] = items ? items.length : 0;
+
+    /**
+     * The completed feature count.
+     * @type {number}
+     * @protected
+     */
+    this['completed'] = 0;
+
+    // Listen for events in the capture phase. The exporter is disposed in the complete/error event handler in
+    // ExportManager, which is registered before these and will remove all listeners.
+    this.exporter.listen(ThreadEventType.PROGRESS, this.onProgress, true, this);
+    this.exporter.listen(OSEventType.COMPLETE, this.onComplete, true, this);
+    this.exporter.listen(OSEventType.ERROR, this.onError, true, this);
+
+    $scope.$emit(WindowEventType.READY);
+  }
+
+  /**
+   * Angular $onDestroy lifecycle call.
+   */
+  $onDestroy() {
+    this.exporter.unlisten(ThreadEventType.PROGRESS, this.onProgress, true, this);
+    this.exporter.unlisten(OSEventType.COMPLETE, this.onComplete, true, this);
+    this.exporter.unlisten(OSEventType.ERROR, this.onError, true, this);
+  }
+
+  /**
+   * Handler for progress events. Updates the progress bar.
+   * @param {ThreadProgressEvent} event
+   * @export
+   */
+  onProgress(event) {
+    const loaded = event.getLoaded();
+    const total = event.getTotal();
+    this['completed'] = 100 * loaded / total;
+    apply(this.scope);
+  }
+
+  /**
+   * Handler for complete events. Closes the window.
+   * @param {GoogEvent} event
+   * @export
+   */
+  onComplete(event) {
+    close(this.element);
+  }
+
+  /**
+   * Handler for error events. Closes the window.
+   * @param {GoogEvent} event
+   * @export
+   */
+  onError(event) {
+    close(this.element);
+  }
+}
+
+
+/**
+ * Window ID.
+ * @type {string}
+ */
+const WINDOW_ID = 'exportProgress';
+
+
+/**
+ * Launches the export progress window for a given exporter.
+ * @param {IExportMethod} exporter
+ */
+export const launch = (exporter) => {
+  var scopeOptions = {
+    'exporter': exporter
+  };
+
+  var container = angular.element(windowSelector.CONTAINER);
+  var x = container.width() - 350;
```

Comment:
Windows treat negative `x` values as offsets from the right edge of the screen, so you could do `'x': -50` here.

Reply:
Huh, always thought that did the offset from the left side of the window. Didn't think it accounted for the width properly.